### PR TITLE
Use datepicker instead of time slider

### PR DESCRIPTION
**Pull request by annarieger**
With this PR we get rid of the previously used time slider to determinate the time range for search. Instead of this two datepickers (for start and end search date) will be used now (s. [uib-datepicker-popup](https://angular-ui.github.io/bootstrap/#/datepickerPopup) for further details).

This PR coveres the https://github.com/terrestris/SolrHeatmap/issues/19 issue.

Please review @terrestris/owners

### CODE REVIEW

File: app/service/HeatMapSourceGenerator.js [hunk 1]
```diff
@@ -6,8 +6,8 @@ angular
     .factory('HeatMapSourceGenerator', ['Map', '$rootScope', '$filter', '$http', function(MapService, $rootScope, $filter, $http) {
 
         var searchObj = {
-            yearMin: 2005,
-            yearMax: 2016,
+            minDate: new Date('2000-01-01'),
+            maxDate: new Date('2016-12-31'),
```
Comment (marcjansen) [inline app/service/HeatMapSourceGenerator.js:10]: Should this be the last day of the current year?

File: app/service/HeatMapSourceGenerator.js [hunk 2]
```diff
@@ -144,4 +149,13 @@ angular
...
         }
+
+        /**
+         * Returns the formatted date object that can be parsed by API.
+         * @param {Date} date full date object (e.g. 'Sat Jan 01 2000 01:00:00 GMT+0100 (CET))
+         * @return {String} formatted date as string (e.g. '2000-01-01')
+         */
+         function getFormattedDateString(date){
+             return date.getFullYear() + "-" + ("0" + (date.getMonth() + 1)).slice(-2) + "-" + ("0" + date.getDate()).slice(-2);
+         }
```
Comment (marcjansen) [inline app/service/HeatMapSourceGenerator.js:160]: Number padding with zeros could be a util.

File: build/hm-client.js [hunk 3]
```diff
@@ -70,6 +70,94 @@ angular
...
 
 /**
+ * DatePickerCtrl Controller
+ */
+angular.module('SolrHeatmapApp')
+    .controller('DatePickerCtrl', ['HeatMapSourceGenerator', '$scope', function(HeatMapSourceGeneratorService, $scope) {
+
+        $scope.initialDateOptions = {
+            minDate: new Date('2000-01-01'),
+            maxDate: new Date('2016-12-31')
```
Comment (marcjansen) [inline build/hm-client.js:80]: See comment above.

File: build/hm-client.js [hunk 4]
```diff
@@ -239,8 +327,8 @@ angular
     .factory('HeatMapSourceGenerator', ['Map', '$rootScope', '$filter', '$http', function(MapService, $rootScope, $filter, $http) {
 
         var searchObj = {
-            yearMin: 2005,
-            yearMax: 2016,
+            minDate: new Date('2000-01-01'),
+            maxDate: new Date('2016-12-31'),
```
Comment (marcjansen) [inline build/hm-client.js:331]: Dito.

**Comment by marcjansen** [comment]
Some minor comments, otherwise looking good to me. Hat-tip for adding docs.

File: app/service/HeatMapSourceGenerator.js (hunk 1 above)
Comment (annarieger) [inline app/service/HeatMapSourceGenerator.js:10]: Yes, both date limits (min/max) were discussed with the customer.

File: build/hm-client.js (hunk 3 above)
Comment (annarieger) [inline build/hm-client.js:80]: See answer above :yum:

**Comment by annarieger** [comment]
Thanks a lot for your review, @marcjansen! Your remarks will be adressed ASAP.

**Pull request merged**